orangecontrib/text/widgets/owduplicates.py

```diff
@@ -198,21 +188,12 @@ def send_corpus(self):
                 'Duplicates Cluster',
                 values=[str(Cluster(v)) for v in set(self.clustering_mask.flatten())]
             )
-            corpus, domain = self.corpus, self.corpus.domain
-            attrs = domain.attributes
-            class_ = domain.class_vars
-            metas = domain.metas
-
-            if self.cluster_role == self.AttributeRole:
-                attrs = attrs + (cluster_var,)
-            elif self.cluster_role == self.ClassRole:
-                class_ = class_ + (cluster_var,)
-            elif self.cluster_role == self.MetaRole:
-                metas = metas + (cluster_var,)
-
-            domain = Domain(attrs, class_, metas)
-            corpus = corpus.from_table(domain, corpus)
-            corpus.get_column_view(cluster_var)[0][:] = self.clustering_mask
+            d = self.corpus.domain
+            domain = Domain(d.attributes, d.class_vars, d.metas + (cluster_var,))
+
+            corpus = self.corpus.transform(domain)
```

Comment:
`Table.transform` by itself would not guarantee a copy (unlockable table). But as you added a new meta column, that part is certainly going to be unlockable. So this is OK.

Reply:
I added a comment that is understandable why it is done that way. The same principle is also used in Orange's k-means widget.